ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js

```diff
@@ -212,23 +212,37 @@ module.exports = Backbone.AssociatedModel.extend({
             this.updateResultCountsBySource(this.createIndexOfSourceToResultCount(metacardIdToSourcesIndex, resp.results));
         }
 
+        this.addQueuedResults(resp.results);
+
+        if (this.get('queuedResults').fullCollection.length !== 0) {
+            // merges the remaining queued results not from the cache
+            this.mergeQueue(true, false);
+        }
+
         return {
-            queuedResults: resp.results,
+            queuedResults: [],
             results: [],
-            status: resp.status
+            status: resp.status,
+            merged: this.get('merged') === false ? false : resp.results.length === 0
         };
     },
+    // we have to do a reset because adding is so slow that it will cause a partial merge to initiate
+    addQueuedResults(results) {
+      const existingQueue = this.get('queuedResults').fullCollection.models;
+      this.get('queuedResults').fullCollection.reset(
+          existingQueue.concat(results)
+      );
+    },
     allowAutoMerge: function () {
         if (this.get('results').length === 0 || !this.get('currentlyViewed')) {
             return true;
         } else {
             return (Date.now() - this.lastMerge) < properties.getAutoMergeTime();
         }
     },
-    mergeQueue: function (userTriggered) {
+    mergeQueue: function (userTriggered, includeQueuedCache = true) {
```

Comment:
I think we need to look into why we need the logic for the queued cache.  I'm hoping we can remove it from master.

Reply:
I can remove the part with the queued cache results and come up with an alternative solution for that.